### ISSUE TITLE
[Train] Get train_func return value

### ISSUE THREAD

**Issue by Hxinyue** [ray-project/ray#49707]
### What happened + What you expected to happen

When ray train is performed, the train_loop_per_worker function has a return value. How can I get the return value of the train_loop_per_worker function after running the trainer.fit()

### Versions / Dependencies

ray==2.11.0
python==3.10

### Reproduction script

```python
def main(config):
    alg_id = config["alg_id"]
	return {"success": 1}

trainer = TorchTrainer(
	main,
	scaling_config=ScalingConfig(use_gpu=use_gpu, num_workers=num_workers,
								 resources_per_worker={"CPU": cpu, "GPU": gpu}),
	train_loop_config=config,
	...
)
trainer.fit()
```

### Issue Severity

High: It blocks me from completing my task.

**Comment by matthewdeng** [comment]
Are you able to report the results as a metric and/or checkpoint?

**Comment by Hxinyue** [comment]
Yes, I have tried the following methods and can obtain the return value.
`def train_func():`
`    import ray.train`
`    metrics = {"res": output_params}`
`    ray.train.report(metrics)`
`result = trainer.fit()`
`output_params = result.metrics["res"]`
